### PR TITLE
add go mod to test to enable import from library repo

### PR DESCRIPTION
**Pull request by mmulholla**
Add go.mod for the test directory to enable library repository tests to import from it.

### CODE REVIEW

**Review by amisevsk — COMMENTED** [review]
I'm not sure I understand why this is being done (having a submodule that imports its parent). We're basically duplicating all of our code in `test/vendor`

**Comment by mmulholla** [comment]
@amisevsk the library repository needs to import from test/v200/utils/common as that is shared code with the library repository tests

**Comment by amisevsk** [comment]
> the library repository needs to import from test/v200/utils/common as that is shared code with the library repository tests

Can't the library just import it from `github.com/devfile/api/v2/test/v200/utils/common`?

**Comment by mmulholla** [comment]
> > the library repository needs to import from test/v200/utils/common as that is shared code with the library repository tests
> 
> Can't the library just import it from `github.com/devfile/api/v2/test/v200/utils/common`?

Not unless the tests are moved under the pkg directory. What I added is similar to the /api/generator directory

**Review by maysunfaisal — COMMENTED** [review]
not sure why we're vendoring the repo, is there a reason?

**Comment by mmulholla** [comment]
> not sure why we're vendoring the repo, is there a reason?

Just copied what other directories with go.mod do

**Comment by mmulholla** [comment]
I am held up by this, please approve, if you can't approve we better go back and revisit the generator directory.

**Comment by mmulholla** [comment]
@maysunfaisal the vendor directory has been removed.

File: test/v200/utils/api/test_utils.go [hunk 1]
```diff
@@ -9,7 +9,7 @@ import (
 	"strings"
 	"testing"
 
-	commonUtils "github.com/devfile/api/v2/test/v200/utils/common"
+	commonUtils "github.com/mmulholla/api/test/v200/utils/common"
```
Comment (amisevsk) [inline test/v200/utils/api/test_utils.go:12]: This needs cleanup before merge if nothing else.

**Comment by amisevsk** [comment]
Creating a new project with the go mod file
```
module test

go 1.14

require github.com/devfile/api/v2 v2.0.0-20210223145532-81859eaef987 // the current main commit
```
allows me to e.g. import commonUtils no problem:

```go
package main

import (
	commonUtils "github.com/devfile/api/v2/test/v200/utils/common"
)

func main() {
	commonUtils.GetDevFileName()
}
```

No idea why this isn't importing automatically, I suspect it's a stale goproxy.

**Comment by amisevsk** [comment]
Playing with it a bit more, running `go get -u github.com/devfile/api/v2@81859eaef987` in the library repo will fix the problem.

File: test/v200/utils/api/test_utils.go (hunk 1 above)
Comment (mmulholla) [reply test/v200/utils/api/test_utils.go]: oops, was using the same branch for testing. Fixed now

**Comment by mmulholla** [comment]
> Playing with it a bit more, running `go get -u github.com/devfile/api/v2@81859eaef987` in the library repo will fix the problem.

Thx, I switched to a new command window and that seem to fix the library issue importing validation too.

**Comment by mmulholla** [comment]
> Creating a new project with the go mod file
> 
> ```
> module test
> 
> go 1.14
> 
> require github.com/devfile/api/v2 v2.0.0-20210223145532-81859eaef987 // the current main commit
> ```
> 
> allows me to e.g. import commonUtils no problem:
> 
> ```go
> package main
> 
> import (
> 	commonUtils "github.com/devfile/api/v2/test/v200/utils/common"
> )
> 
> func main() {
> 	commonUtils.GetDevFileName()
> }
> ```
> 
> No idea why this isn't importing automatically, I suspect it's ~a stale goproxy.~ because the highest tagged release is https://github.com/devfile/api/releases/tag/v2.0.0, from Jan 18th. Making a submodule will have the same problem until a new tag is created or consumers manually set the version via commit (which is what we do in the DevWorkspace Operator)

OK, I have changed the module in go.sum to be just "test" and locally import from "api/v2/test" and everything appears to work for the tests in the api repository. Not sure I can test import into the library repository until this is merged.

**Comment by mmulholla** [comment]
After discussions with Angel (thank you) it appears that this PR is redundant, the api test directory does not need to create a module for the library repository to import it. Testing now, but closing this PR.

**Pull request closed without merging**